crates/app/src/locked.rs

```diff
@@ -126,6 +126,9 @@ pub struct ContentRef {
     /// If set, the content must have the given SHA-256 digest.
     #[serde(default, skip_serializing_if = "Option::is_none")]
     pub digest: Option<String>,
+    /// If the content represents a compressed archive.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub archive: Option<bool>,
```

Comment:
I'd really like to avoid having this in the lockfile; I think ultimately it would keep things simpler. That _does_ require some refactoring in `Client::pull` but I actually want to make some other changes in there soon so I'm happy to tackle that if you'd like.

Reply:
Sure, we can remove.  Were you thinking we'd differentiate based on directory where we pull layers into eg we currently parse the media type and store content in wasm or data dirs today... would we add an archive dir or similar?

Reply:
We can probably just stick them in the data dir. Even the split between the wasm and data dirs isn't _really_ necessary.